### PR TITLE
Downloading only the script folder

### LINKED ISSUE #283: Downloading only script folder from Meilix Generator
Now the chroot script downloads all Generator repo and then work only with script folder. 
The idea is to download only the script folder.

### PR DESCRIPTION
**Pull request by meets2tarun**
### Short description
I have:
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`

### For the reviewers
I have:
- [ ] Reviewed this pull request by an authorized contributor.
- [ ] The reviewer is assigned to the pull request.

Fixes #283

### CODE REVIEW

File: meilix-default-settings/etc/skel/.dmrc [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[Desktop]
```
Comment (xeon-zolt) [inline meilix-default-settings/etc/skel/.dmrc]: should be part of another pull request

File: scripts/chroot.sh [hunk 2]
```diff
@@ -155,10 +155,9 @@ apt-get install -f
 apt-get -qq -y remove dconf-tools
 
 # Instalation of packages from Generator Webapp
-SCRIPT_URL=https://www.github.com/fossasia/meilix-generator/archive/master.zip
-wget -O $scripts.zip $SCRIPT_URL
-unzip scripts.zip
-SCRIPTS_FOLDER_IN_ZIP="meilix-generator-master/scripts"
+SCRIPT_URL=https://github.com/fossasia/meilix-generator/trunk/scripts
```
Comment (xeon-zolt) [inline scripts/chroot.sh]: ok it will work just in case multiple users are not using the generator at once

File: scripts/chroot.sh [hunk 3]
```diff
@@ -155,10 +155,9 @@ apt-get install -f
...
+svn checkout $SCRIPT_URL
```
Comment (xeon-zolt) [inline scripts/chroot.sh]: why are you using a VCS here ? for downloading ?

File: meilix-default-settings/etc/skel/.dmrc (hunk 1 above)
Comment (meets2tarun) [reply meilix-default-settings/etc/skel/.dmrc]: Yes, there is a different PR already.
https://github.com/fossasia/meilix/pull/282

File: scripts/chroot.sh (hunk 2 above)
Comment (meets2tarun) [reply scripts/chroot.sh]: Yes, it seems similar to me too

File: scripts/chroot.sh (hunk 3 above)
Comment (meets2tarun) [reply scripts/chroot.sh]: Because I want to download at particular script folder only not the whole repo.

**Comment by meets2tarun** [comment]
@niccokunzmann 
Please review https://github.com/fossasia/meilix/pull/284/files#diff-333f2f40be6e16a3a754ae82087c11fb through this, only the script folder will get downloaded.

File: scripts/chroot.sh (hunk 3 above)
Comment (niccokunzmann) [reply scripts/chroot.sh]: I think, we can change it do DownGit: https://stackoverflow.com/a/38879691 Please check if this is possible. The purpose of this code is to allow the download as a zip file, so we can change it to almost any location because people know how to create zip files.

**Comment by meets2tarun** [comment]
@niccokunzmann 
Please review, I have added the changes to only download the script folder zip.

**Comment by meets2tarun** [comment]
@xeon-zolt 
Please review

**Comment by meets2tarun** [comment]
@niccokunzmann @tabesin 
Shall we merge this?

File: scripts/chroot.sh [hunk 4]
```diff
@@ -165,10 +165,10 @@ apt-get install -f
 apt-get -qq -y remove dconf-tools
 
 # Instalation of packages from Generator Webapp
-SCRIPT_URL=https://www.github.com/fossasia/meilix-generator/archive/master.zip
+SCRIPT_URL=https://minhaskamal.github.io/DownGit/#/home?url=https://github.com/fossasia/meilix-generator/tree/master/scripts
```
Comment (niccokunzmann) [inline scripts/chroot.sh]: This url does not work.. 

```
$ wget -O- https://minhaskamal.github.io/DownGit/#/home?url=https://github.com/fossasia/meilix-generator/tree/master/scripts 
--2018-07-09 19:38:08--  https://minhaskamal.github.io/DownGit/
Resolving minhaskamal.github.io (minhaskamal.github.io)... 185.199.109.153, 185.199.111.153, 185.199.108.153, ...
Connecting to minhaskamal.github.io (minhaskamal.github.io)|185.199.109.153|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 2026 (2.0K) [text/html]
Saving to: ‘STDOUT’

-                     0%[                      ]       0  --.-KB/s             <!DOCTYPE html>
<html ng-app="downGit">
<head>
    
    <!-------------------------------- META -------------------------------->
    
    <meta charset="utf-8"/>
    <meta name="keywords" content="GitHub Directory Link, GitHub Directory, Github Link, GitHub Directory Download, GitHub, Download"/>
    <meta name="description" content="Create GitHub Directory Download Link"/>
    <meta name="author" content="Minhas Kamal"/>
    
    
    <title>DownGit</title>
    <link rel="icon" type="image/ico" href="res/images/downgit.png">
    
    
    <!-------------------------------- LINK -------------------------------->
    
    <link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-1q8mTJOASx8j1Au+a5WDVnPi2lkFfwwEAa8hDDdjZlpLegxhjVME1fgjWPGmkzs7" crossorigin="anonymous" />
    <link href="lib/angular-toastr(2.0.0).min.css" rel="stylesheet"/>
    
    <link href="app/site.css" rel="stylesheet" />
</head>
<body>
    
    <div ng-view class=""></div>
    
    
    <!------------------------------- SCRIPT ------------------------------>
    
    <script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.5.6/angular.min.js"></script>
    <script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.5.6/angular-route.min.js"></script>
    <script src="https://code.jquery.com/jquery-2.2.4.min.js" integrity="sha256-BbhdlvQf/xTY9gja0Dq3HiwQF8LaCRTXxZKRutelT44=" crossorigin="anonymous"></script>
    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/js/bootstrap.min.js" integrity="sha384-0mSbJDEHialfmuBBQP6A4Qrprq5OVfW37PRR3j5ELqxss1yVqOtnepnHVP9aJ7xS" crossorigin="anonymous"></script>
    <script src="lib/angular-toastr(2.0.0).tpls.min.js"></script>
    <script src="https://cdn.rawgit.com/Stuk/jszip/v3.0.0/dist/jszip.min.js"></script>
    <script src="lib/filesaver.min.js"></script>

    <script src="app/app.js"></script>
    <script src="app/home/home.js"></script>
    <script src="app/home/down-git.js"></script>
</body>
-                   100%[=====================>]   1.98K  --.-KB/s   in 0s     

2018-07-09 19:38:09 (12.7 MB/s) - written to stdout [2026/2026]
````

**Comment by niccokunzmann** [comment]
It seems like downGit creates the zip archive in the browser. It uses the GitHub API to build the zip archive.

**Comment by meets2tarun** [comment]
It tried the URL separately to get the zip file.
Ok, I'm trying again the script.

File: scripts/chroot.sh (hunk 4 above)
Comment (meets2tarun) [reply scripts/chroot.sh]: I'm getting same error with `curl` also.
But the URL successfully downloads the script as zip if we only execute the URL.

Comment (niccokunzmann) [reply scripts/chroot.sh]: The zip file is built in the browser using JavaScript. Wget is not a browser and can not do that.

Comment (tabesin) [reply scripts/chroot.sh]: I have no idea why you download scripts from meilix-generator. This seems to be also a security risk. I would only set variables as switches and have the scripts in meilix.

**Comment by abishekvashok** [comment]
I think we should close this as we r having the scripts in meilix itself.
